test/custom_resource_crud/custom_resource_crud_c3_test.go

```diff
@@ -0,0 +1,296 @@
+
+			PvcName = fmt.Sprintf(testenv.SearchHeadPvcEtc, deployment.GetName(), 0)
+			testenv.VerifyPVCs(deployment, testenvInstance.GetName(), PvcName, SH0EtcPVC, true)
+			PvcName = fmt.Sprintf(testenv.SearchHeadPvcVar, deployment.GetName(), 0)
+			testenv.VerifyPVCs(deployment, testenvInstance.GetName(), PvcName, SH0VarPVC, true)
+
+			// Verify Deployer PVCs (etc and var) exists
+			PvcName = fmt.Sprintf(testenv.DeployerPvcEtc, deployment.GetName(), 0)
+			testenv.VerifyPVCs(deployment, testenvInstance.GetName(), PvcName, DeployerEtcPvc, true)
+			PvcName = fmt.Sprintf(testenv.DeployerPvcVar, deployment.GetName(), 0)
+			testenv.VerifyPVCs(deployment, testenvInstance.GetName(), PvcName, DeployerVarPvc, true)
+
+			// Verify Indexers PVCs (etc and var) have been deleted
+			PvcName = fmt.Sprintf(testenv.IndexerPvcEtc, deployment.GetName(), 2)
+			testenv.VerifyPVCs(deployment, testenvInstance.GetName(), PvcName, IDX2EtcPVC, true)
+			PvcName = fmt.Sprintf(testenv.IndexerPvcVar, deployment.GetName(), 2)
+			testenv.VerifyPVCs(deployment, testenvInstance.GetName(), PvcName, IDX2VarPVC, true)
+
+			PvcName = fmt.Sprintf(testenv.IndexerPvcEtc, deployment.GetName(), 1)
+			testenv.VerifyPVCs(deployment, testenvInstance.GetName(), PvcName, IDX1EtcPVC, true)
+			PvcName = fmt.Sprintf(testenv.IndexerPvcVar, deployment.GetName(), 1)
+			testenv.VerifyPVCs(deployment, testenvInstance.GetName(), PvcName, IDX1VarPVC, true)
+
+			PvcName = fmt.Sprintf(testenv.IndexerPvcEtc, deployment.GetName(), 0)
+			testenv.VerifyPVCs(deployment, testenvInstance.GetName(), PvcName, IDX0EtcPVC, true)
+			PvcName = fmt.Sprintf(testenv.IndexerPvcVar, deployment.GetName(), 0)
+			testenv.VerifyPVCs(deployment, testenvInstance.GetName(), PvcName, IDX0VarPVC, true)
+
+			// Verify Cluster Master PVCs (etc and var) have been deleted
+			PvcName = fmt.Sprintf(testenv.ClusterMasterPvcEtc, deployment.GetName(), 0)
+			testenv.VerifyPVCs(deployment, testenvInstance.GetName(), PvcName, CMEtcPvc, true)
+			PvcName = fmt.Sprintf(testenv.ClusterMasterPvcVar, deployment.GetName(), 0)
+			testenv.VerifyPVCs(deployment, testenvInstance.GetName(), PvcName, CMVarPvc, true)
+
+			// Delete the Search Head Cluster
+			shc := &enterprisev1.SearchHeadCluster{}
+			deployment.GetInstance(deployment.GetName()+"-shc", shc)
+			err = deployment.DeleteCR(shc)
+			Expect(err).To(Succeed(), "Unable to delete SHC instance", "SHC Name", shc)
+
+			// Delete the Indexer Cluster
+			idxc := &enterprisev1.IndexerCluster{}
+			deployment.GetInstance(deployment.GetName()+"-idxc", idxc)
+			err = deployment.DeleteCR(idxc)
+			Expect(err).To(Succeed(), "Unable to delete IDXC instance", "IDXC Name", idxc)
+
+			// Delete the Cluster Master
+			cm := &enterprisev1.ClusterMaster{}
+			deployment.GetInstance(deployment.GetName(), cm)
+			err = deployment.DeleteCR(cm)
+			Expect(err).To(Succeed(), "Unable to delete CM instance", "CM Name", cm)
+
+			// Verify Search Heads PVCs (etc and var) have been deleted
```

Comment:
Would be better to move all the repeated code to a single method. Look at validate `VerifySplunkServerConfSecrets` 
for example